src/guides/v2.3/extension-dev-guide/framework/float-comparator.md

```diff
@@ -0,0 +1,40 @@
+---
+group: php-developer-guide
+title: Float Comparator
+contributor_name: Adarsh Manickam
+contributor_link: https://github.com/drpayyne
+---
+
+## Overview
+
+The [`Magento\Framework\Math\FloatComparator`]({{ site.mage2bloburl }}/{{ page.guide_version }}/lib/internal/Magento/Framework/Math/FloatComparator.php){:target="\_blank"} library provides methods to compare float values with an approriate epsilon value (`0.00001`). Floats usually have rounding errors while doing mathematical operations, so the conventional way of comparing two floats using the equality operator (`==`), may not always lead to expected results. For more information, please refer to this [article](https://floating-point-gui.de/errors/comparison/).
+
+## Usage
+
+| Method               | Description                                                                                                                       |
+| -------------------- | --------------------------------------------------------------------------------------------------------------------------------- |
+| `equal`              | Compares two given floats and returns a boolean value describing if they're equal.                                                |
+| `greaterThan`        | Compares two given floats and returns a boolean value describing if the first float is greater than the second float.             |
+| `greaterThanOrEqual` | Compares two given floats and returns a boolean value describing if the first float is greater than or equal to the second float. |
+
+## Examples
+
+For the below examples, let's assume that the `$this->floatComparator` property is an instatiated object of the `FloatComparator` class.
```

Comment:
Hi, @drpayyne !

What do you think about this option of your explanation:

```
In the examples below `$this->floatComparator` is an instatiated object of the [`Magento\Framework\Math\FloatComparator`]({{ site.mage2bloburl }}/{{ page.guide_version }}/lib/internal/Magento/Framework/Math/FloatComparator.php){:target="\_blank"} class.
```

Reply:
Hi thanks for your suggestion. I decided not to link it again since it's already at the top of the page. Didn't feel right to link every occurrence of that class in that page. What do you think?

Reply:
Makes sense :)

What do you think about:

```
In the examples below `$this->floatComparator` is an instatiated object of the `Magento\Framework\Math\FloatComparator` class.
```

Thanks

Reply:
Done, thanks!